impeller/entity/contents/solid_fill_utils.cc

```diff
@@ -4,8 +4,35 @@
 
 #include "impeller/entity/contents/solid_fill_utils.h"
 
+#include "impeller/geometry/path.h"
+#include "impeller/renderer/device_buffer.h"
+#include "impeller/renderer/host_buffer.h"
+#include "impeller/tessellator/tessellator.h"
+
 namespace impeller {
 
-//
+VertexBuffer CreateSolidFillVertices(std::shared_ptr<Tessellator> tessellator,
+                                     const Path& path,
+                                     HostBuffer& buffer) {
+  VertexBuffer vertex_buffer;
+  VertexBuffer* vertex_buffer_out = &vertex_buffer;
+  auto tesselation_result = tessellator->TessellateBuilder(
+      path.GetFillType(), path.CreatePolyline(),
+      [&vertex_buffer_out, &buffer](
+          const float* vertices, size_t vertices_count, const uint16_t* indices,
+          size_t indices_count) {
+        vertex_buffer_out->vertex_buffer = buffer.Emplace(
+            vertices, vertices_count * sizeof(float), alignof(float));
+        vertex_buffer_out->index_buffer = buffer.Emplace(
+            indices, indices_count * sizeof(uint16_t), alignof(uint16_t));
```

Comment:
This seems like it'll have the added benefit of actually giving us indexed vertices instead of our current strategy that unrolls them.